### PR TITLE
GDB-10716: Create Chat Panel Layout

### PR DESCRIPTION
**Pull request by boyan-tonchev**
## What
Updated the chat panel view to improve the layout and structure.

## Why
To meet the updated design requirements.

## How
Created a responsive layout using flexbox to manage the display of chat questions and answers, ensuring the panel matches the specified requirements.
image

### CODE REVIEW

File: src/js/angular/ttyg/templates/chat-panel.html [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
-            <input class="form-control" ng-model="chatItem.question.message" type="text">
-            <button class="btn btn-primary" ng-disabled="!chatItem.question.message" ng-click="ask()">
-                <i class="fa fa-arrow-turn-left-up"></i>
-                {{ 'ttyg.ask.button' | translate }}
-            </button>
-        </div>
+    <div class="mew-question mb-2 d-flex">
+        <input class="form-control question-input" ng-model="chatItem.question.message" type="text">
+        <button class="btn btn-primary" ng-disabled="!chatItem.question.message" ng-click="ask()">
+            <i class="fa fa-arrow-turn-left-up"></i>
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/templates/chat-panel.html:36]: fa-regular

Comment (boyan-tonchev) [reply src/js/angular/ttyg/templates/chat-panel.html]: Done

**Review by svilenvelikov — DISMISSED** [review]

File: src/js/angular/ttyg/controllers/ttyg-view.controller.js [hunk 2]
```diff
@@ -271,11 +262,15 @@ function TTYGViewCtrl($rootScope, $scope, $http, $timeout, $translate, $uibModal
...
      */
     const onAskQuestion = (chatItem) => {
-        const item = cloneDeep(chatItem);
         TTYGService.askQuestion(chatItem)
             .then((answer) => {
-                item.answer = answer;
-                TTYGContextService.emit(TTYGEventName.ASK_QUESTION_SUCCESSFUL, item);
+                const selectedChat = TTYGContextService.getSelectedChat();
+                if (selectedChat && selectedChat.id === chatItem.chatId) {
+                    const item = cloneDeep(chatItem);
```
Comment (svilenvelikov) [inline src/js/angular/ttyg/controllers/ttyg-view.controller.js:269]: why do you need to clone this again

File: test-cypress/integration/ttyg/chat-panel.spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import {ChatPanelSteps} from "../../steps/ttyg/chat-panel-steps";
+
+describe('Ttyg ChatPanel', () => {
+
+    let repositoryId;
+
+    beforeEach(() => {
+        // Create an actual repository to prevent stubbing all background requests that are not related to the ttyg view
+        repositoryId = 'starwars';
+        cy.createRepository({id: repositoryId});
```
Comment (svilenvelikov) [inline test-cypress/integration/ttyg/chat-panel.spec.js:13]: remove this

File: test-cypress/integration/ttyg/chat-panel.spec.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+        RepositoriesStubs.stubRepositories(0, '/repositories/get-ttyg-repositories.json');
+        cy.presetRepository('starwars');
+    });
+
+    afterEach(() => {
+        cy.deleteRepository(repositoryId);
```
Comment (svilenvelikov) [inline test-cypress/integration/ttyg/chat-panel.spec.js:20]: remove it

**Review by svilenvelikov — DISMISSED** [review]

File: test-cypress/integration/ttyg/chat-panel.spec.js (hunk 4 above)
Comment (boyan-tonchev) [reply test-cypress/integration/ttyg/chat-panel.spec.js]: Done

File: test-cypress/integration/ttyg/chat-panel.spec.js (hunk 3 above)
Comment (boyan-tonchev) [reply test-cypress/integration/ttyg/chat-panel.spec.js]: Done

**Review by svilenvelikov — APPROVED** [review]

**Pull request merged**